src/lib/y2partitioner/immediate_unmount.rb

```diff
@@ -0,0 +1,173 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast/i18n"
+require "yast2/popup"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  # Mixin that offers a dialog to immediate unmount a block device
+  #
+  # @note This feature is intended to allow to unmount devices before the commit phase, for example,
+  #   when a device is deleted or resized. But there are serveral scenarios where the actions require
+  #   unmount the device and the user is not asked for it. For instance, when a new partition table
+  #   is created over a disk and some of its partitions are already mounted.
+  module ImmediateUnmount
+    include Yast::Logger
+    include Yast::I18n
+
+  private
+
+    # Sets textdomain
+    def included(_target)
+      textdomain "storage"
+    end
+
+    # Shows a Popup dialog to try to unmount the device
+    #
+    # @param device [Y2Storage::BlkDevice]
+    # @param full_message [String, nil] message to show in the dialog (param note would be ignored)
+    # @param note [String, nil] note to add to the generic message (ignored if full_message is used)
+    # @param allow_continue [Boolean] if it should allow to continue without unmounting
+    #
+    # @return [Boolean]
+    def immediate_unmount(device, full_message: nil, note: nil, allow_continue: true)
```

Comment:
To be honest, I do not like now design when I see it whole as only method that makes sense for users of this mixin is exactly this one. So what I propose? Having class ( e.g. inner class for unmount that gets params on instance and do action ). This way it allows to:

1. not pass allow_continue everywhere
2. not need that included trick for translations
3. easier test it
4. if makes sense ( now it is a bit on edge ) create subclasses ( e.g. for allow continue versus not one ).